src/v3/src/components/ConsentHeader/ConsentHeader.tsx

```diff
@@ -0,0 +1,169 @@
+  const appName = escape(label);
+  const granularConsentTitle = loc(
+    'oie.consent.scopes.granular.title',
+    'login',
+    [appName],
+    {
+      $1: { element: 'h2', attributes: { class: 'no-translate' } },
+      $2: { element: 'p' },
+    },
+  );
+  const parsedGranularConsentTitle = useHtmlContentParser(
+    granularConsentTitle,
+    { replace: getHeadingReplacerFn({}, 'h2', 2, 3) },
+  );
+
+  if (!idxTransaction?.nextStep || !CONSENT_HEADER_STEPS.includes(idxTransaction.nextStep.name)) {
+    return null;
+  }
+  const stepName = idxTransaction.nextStep.name;
+
+  const getAppLogo = (altText: string, logoHref?: string) => (
+    typeof logoHref !== 'undefined'
+      ? (
+        <Box
+          component="img"
+          src={logoHref}
+          width="32px"
+          height="32px"
+          alt={altText}
+          className="client-logo custom-logo"
+          aria-hidden="true"
+        />
+      )
+      : (
+        // TODO: This is a temporary icon used until UX provides one
+        <SettingsIcon
+          titleAccess={altText}
+          classes="client-logo default-logo"
+          sx={{ width: '32px !important', height: '32px !important' }}
+          aria-hidden
+        />
+      )
+  );
+
+  const getAppIcon = () => {
+    const href: string | undefined = (clientUri as Record<string, string | undefined>)?.href;
+    const logoHref: string | undefined = (logo as Record<string, string | undefined>)?.href;
+    const altText = loc('logo.for.the.app.alt.text', 'login');
+    return (
+      <Box
+        display="flex"
+        justifyContent="center"
+      >
+        {typeof href !== 'undefined'
+          ? (
+            <Box component="div">
+              <Link
+                href={href}
+                target="_blank"
+                aria-label={altText}
+              >
+                {getAppLogo(altText, logoHref)}
+              </Link>
+            </Box>
+          )
+          : getAppLogo(altText, logoHref)}
+      </Box>
+    );
+  };
+
+  const getHeaderContent = () => {
+    if ([IDX_STEP.CONSENT_ADMIN, IDX_STEP.CONSENT_ENDUSER].includes(stepName)) {
+      // @ts-expect-error OKTA-598777 authentication missing from IdxContext interface
+      const { context: { authentication: { value: { issuer } } = {} } } = idxTransaction;
+      const hasIssuer = stepName === IDX_STEP.CONSENT_ADMIN && typeof issuer?.uri !== 'undefined';
+      const titleText = stepName === IDX_STEP.CONSENT_ADMIN
+        ? loc('oie.consent.scopes.admin.title', 'login')
+        : loc('oie.consent.scopes.enduser.title', 'login');
+      return (
+        <Fragment>
+          <Box
+            component="span"
+            className="title-text"
+            textAlign="center"
+          >
+            <Typography
+              component="h1"
+              variant="h3"
+              className="no-translate"
+            >
+              {appName}
+              &nbsp;
```

Comment:
Curious why the nbsp is needed here

Reply:
We need a space between teh app name and the Title, otherwise the content runs together, I changed this to use `{' '}` instead since in the vite branch, there is a lint rule disallowing &nbsp;